### PR TITLE
dont recompute revoked flag if we hit the chat body cache CORE-7266

### PR DESCRIPTION
**Pull request by mmaxim**
This `updateMessages` code in `HybridConversationSource` slows down the speed of the service returning a cache hit for messages successfully stored. It doesn't seem necessary to recompute this flag from what we have in the cache, and results in a load user call to the API server, making it very slow (on the order of a second). I'd like for this to be instantaneous. 

Also, for what it's worth, this code had no tests, and the frontend in its current state "caches" messages in Redux, so I don't think there is much value lost.

cc @maxtaco  in case you disagree.

### CODE REVIEW

File: go/chat/storage/storage.go [hunk 1]
```diff
@@ -297,20 +297,21 @@ type MergeResult struct {
 	DeletedHistory bool
 }
 
-// msgs must be sorted by descending message ID
+// Merge requires msgs must be sorted by descending message ID
```
Comment (mlsteele) [inline go/chat/storage/storage.go]: must be -> to be

same below

File: go/chat/convsource.go [hunk 2]
```diff
@@ -939,7 +874,7 @@ func (s *HybridConversationSource) GetMessagesWithRemotes(ctx context.Context,
 	}
 	if len(merges) > 0 {
 		sort.Sort(ByMsgID(merges))
-		if err = s.mergeMaybeNotify(ctx, convID, uid, merges); err != nil {
+		if err := s.mergeMaybeNotify(ctx, convID, uid, merges); err != nil {
```
Comment (mlsteele) [inline go/chat/convsource.go]: Curious why this change?

Comment (mmaxim) [reply go/chat/convsource.go]: I changed the return type of `FetchMessages` to be `storage.Error`, which failed here.

**Comment by mmaxim** [comment]
Hah nice catch on that comment!

File: go/chat/convsource.go (hunk 2 above)
Comment (mlsteele) [reply go/chat/convsource.go]: Do you mean `GetMessagesWithRemotes`, which is where this line is?
